pkg/crd/gen.go

```diff
@@ -85,6 +85,13 @@ type Generator struct {
 
 	// Year specifies the year to substitute for " YEAR" in the header file.
 	Year string `marker:",optional"`
+
+	// PreserveUnknownFields indicates whether or not we should turn off pruning.
+	//
+	// Specifies spec.preserveUnknownFields value that is false and omitted by default.
+	//
+	// It's required to be false for v1 CRDs.
+	PreserveUnknownFields *bool `marker:",optional"`
```

Comment:
```suggestion
	// DeprecatedV1beta1CompatibilityPreserveUnknownFields indicates whether 
	// or not we should turn off field pruning for this resource.
	//
	// Specifies spec.preserveUnknownFields value that is false and omitted by default.
	// This value can only be specified for CustomResourceDefinitions that were created with
	// `apiextensions.k8s.io/v1beta1`.
	//
	// The field can be set for compatiblity reasons, although strongly discouraged, resource
	// authors should move to a structural OpenAPI schema instead.
	//
	// See https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#field-pruning 
	// for more information about field pruning and v1beta1 resources compatibility.
	DeprecatedV1beta1CompatibilityPreserveUnknownFields *bool `marker:",optional"`
```

Something like that?

Reply:
Is this necessary? By default the marker is not used, it will be used by those who solve specific problem and know what they want.

Reply:
The link is useful as well as different name from previously existing marker, fixed.